apps/finance/app/src/app-state-reducer.js

```diff
@@ -21,13 +21,6 @@ function appStateReducer(state) {
           ...balance,
           amount: new BN(balance.amount),
           decimals: new BN(balance.decimals),
```

Comment:
We can do this later, but now I'm wondering if we should prefer leaving `decimals` as a number (e.g. in the `use-token` data types).

It's usually encoded as `uint8`, so it can never be imprecise, and is _always_ a formatting tool (its only purpose is to be one). It may still be operated on, e.g. with uniswap/conversion APIs, but always just to calculate how many places a decimal should be moved left/right.

Reply:
Yes, I guess the problem will disappear when we will only pass it once to the `TokenAmount` constructor.

I tend to use a `BigInt` (or here a `BN`) by default (e.g. in [`formatTokenAmount()`](https://github.com/aragon/aragon-ui/blob/2c63df362af123a34675f679a8f5b758fa11991a/src/utils/format.js#L69)), but only because the amount is also a `BigInt`, and we can’t calculate a `Number` with a `BigInt`.

Reply:
> Yes, I guess the problem will disappear when we will only pass it once to the TokenAmount constructor.

Yes, agreed!

I mention it here because this app is the only one of ours that uses only numbers for the decimals in some places (in the `tokens` data).